### DISCUSSION TITLE
A Pytest plugin that launches a Text User Interface after a test session

### DISCUSSION THREAD

**Discussion by jeffwright13** [pytest-dev/pytest#9847] [Show and tell]
Go ahead, tear it apart. :-)  Or, just offer constructive criticism - if that's your game. 

https://pypi.org/project/pytest-tui

**Comment by nicoddemus** [comment]
Haven't tried it yet, but looks great! Thanks for sharing!

**Comment by jeffwright13** [comment]
Updated to v0.9.0 in case anyone wants to have a go

**Comment by jeffwright13** [comment]
[New version 0.9.7](https://pypi.org/project/pytest-tui/0.9.7/) includes HTML report generation and autolaunch. My main motivation here was to address a shortcoming in `pytest-html` where the initial `=== test session starts ===` and `=== live log ===` information was not being displayed.

**Reply by jeffwright13** [reply]
[Version 1.3.0](https://github.com/jeffwright13/pytest-tui) features a much-improved HTML report, and revamped internal logic to more accurately report results. A true GUI and a new TUI (featuring the much-anticipated CSS version of Textual) will be next.
